src/components/multi_select_state.js

```diff
@@ -22,21 +22,23 @@ const withMultiSelectState = WrappedComponent =>
       this.handleChange = this.handleChange.bind(this);
       this.getList = this.getList.bind(this);
       this.onKeyUp = this.onKeyUp.bind(this);
+      this.filterDestinationItems = this.filterDestinationItems.bind(this);
 
       const { items, selectedItems } = props;
       this.state = {
         selectedItems,
         items,
-        filteredItems: items
+        filteredItems: items,
```

Comment:
@MaksimDrobchak please go over this file extremely carefully and check all usages of existing "filteredItems" state field. I think you missed to add handling for your new state field in isAllSelected.

Also, please test on your changes the following case:
1. Select several items
2. In Destination section filter items
3. Select more items in Source section - both matching and not matching Destination section filter